app/routes.py

```diff
@@ -1 +1,227 @@
-from flask import Blueprint
\ No newline at end of file
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, make_response, request, abort
+from sqlalchemy import asc, desc 
+import datetime
+import requests
+import os 
+from dotenv import load_dotenv
+
+load_dotenv()
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["POST"])
+def create_tasks():
+    request_body = request.get_json()
+    new_task = Task.from_dict(request_body) 
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    task_dict = {}
+    task_dict["task"] = {"id":new_task.task_id, "title": new_task.title,"description":new_task.description, "is_complete":bool(new_task.completed_at)}
+
+    return make_response(jsonify(task_dict), 201) 
+
+@tasks_bp.route("", methods=["GET"])
+def get_all_tasks():
+    sort_query = request.args.get("sort")
+    if sort_query:
+        if "asc" in sort_query: 
+            tasks = Task.query.order_by(Task.title.asc())
+        elif "desc" in sort_query: 
+            tasks = Task.query.order_by(Task.title.desc()) 
+    else: 
+        tasks = Task.query.all()
+
+    tasks_response = [task.to_dict() for task in tasks]
+    return jsonify(tasks_response) 
+    
+
+def validate_model(cls, model_id):
+    try: 
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message": f"{cls.__name__} {model_id} invalid"}, 400)) 
+
+    model = cls.query.get(model_id)
+    
+    if not model:
+        abort(make_response({"message": f"{cls.__name__} {model_id} not found"}, 404)) 
+
+    return model
```

Comment:
I also know you pointed out that you were concerned about putting all of these routes in the same file, and yes, I would probably organize the code that way, but let me know that then causes a bit of a tricky situation here with `validate_model` which is used by both sets of routes.

My recommendation would be moving it to shared utilities module, something like `utils.py`. This is a very common pattern and while by no means a big module of random utility methods is ideal class design, it often is the least bad option.

Another example of something I might also move to `utils.py` would be the Slack integration code, as that is another one of those "cross-cutting concerns" that doesn't directly relate to one set of routes, but instead could relate to several, presuming a fuller integration implementation.

Though, if that got big enough, I would probably end up with two separate utilities modules, `model_utils.py` for `validate_model` and `slack_utils.py` for the Slack integration.